XPlat/Shopify/iOS/source/Shopify.iOS/ApiDefinition.cs

```diff
@@ -0,0 +1,1964 @@
+
+		[Export ("variantWithOptions:")]
+		ProductVariant GetVariant (OptionValue[] options);
+
+		[Export ("isDefaultVariant")]
+		bool IsDefaultVariant ();
+
+	}
+
+	[BaseType (typeof(UITableViewCell), Name = "BUYProductDescriptionCell")]
+	interface ProductDescriptionCell : Themeable
+	{
+		[Export ("setDescriptionHTML:")]
+		void SetDescriptionHTML (string html);
+
+	}
+
+	[BaseType (typeof(UITableViewCell), Name = "BUYProductHeaderCell")]
+	interface ProductHeaderCell : Themeable
+	{
+		[Export ("setProductVariant:withCurrencyFormatter:")]
+		void SetProductVariant (ProductVariant productVariant, NSNumberFormatter currencyFormatter);
+
+	}
+
+	[BaseType (typeof(UICollectionViewCell), Name = "BUYProductImageCollectionViewCell")]
+	interface ProductImageCollectionViewCell
+	{
+		[Export ("productImageView", ArgumentSemantic.Strong)]
+		ImageView ProductImageView { get; set; }
+
+		[Export ("productImageViewConstraintHeight", ArgumentSemantic.Strong)]
+		NSLayoutConstraint ProductImageViewConstraintHeight { get; set; }
+
+		[Export ("productImageViewConstraintBottom", ArgumentSemantic.Strong)]
+		NSLayoutConstraint ProductImageViewConstraintBottom { get; set; }
+
+		[Export ("setContentOffset:")]
+		void SetContentOffset (CGPoint offset);
+
+	}
+
+	[BaseType (typeof(Object), Name = "BUYProductVariant")]
+	interface ProductVariant
+	{
+		[Export ("product", ArgumentSemantic.Strong)]
+		Product Product { get; set; }
+
+		[Export ("title")]
+		string Title { get; }
+
+		[Export ("options", ArgumentSemantic.Copy)]
+		OptionValue[] Options { get; }
+
+		[Export ("price", ArgumentSemantic.Strong)]
+		NSDecimalNumber Price { get; }
+
+		[Export ("compareAtPrice", ArgumentSemantic.Strong)]
+		NSDecimalNumber CompareAtPrice { get; }
+
+		[Export ("grams", ArgumentSemantic.Strong)]
+		NSDecimalNumber Grams { get; }
+
+		[Export ("requiresShipping", ArgumentSemantic.Strong)]
+		NSNumber RequiresShipping { get; }
+
+		[Export ("sku", ArgumentSemantic.Strong)]
+		string Sku { get; }
+
+		[Export ("taxable", ArgumentSemantic.Strong)]
+		NSNumber Taxable { get; }
+
+		[Export ("position", ArgumentSemantic.Strong)]
+		NSNumber Position { get; }
+
+		[Export ("available")]
+		bool Available { get; }
+
+		// interface BUYProductVariant_Options
+		[Static]
+		[Export ("filterProductVariants:forOptionValue:")]
+		ProductVariant[] FilterProductVariants (ProductVariant[] productVariants, OptionValue optionValue);
+
+	}
+
+	[Category]
+	[BaseType (typeof(ProductVariant), Name = "BUYProductVariant")]
+	interface ProductVariantOptionsExtensions
+	{
+		[Export ("optionValueForName:")]
+		OptionValue OptionValueForName (string optionName);
+
+	}
+
+	[BaseType (typeof(UITableViewCell), Name = "BUYProductVariantCell")]
+	interface ProductVariantCell : Themeable
+	{
+		[Export ("setOptionsForProductVariant:")]
+		void SetOptionsForProductVariant (ProductVariant productVariant);
+
+	}
+
+	[BaseType (typeof(UIView), Name = "BUYProductView")]
+	interface ProductView
```

Comment:
Checking with header, I would suggest to add [DisableDefaultCtor] attribute because it would make it easy to configure product if developer just use Constructor (CGRect rect, Product product, Theme theme, bool showApplePaySetup).